services/src/api/controllers/internships.ctrl.ts

```diff
@@ -0,0 +1,161 @@
+                return res.send({
+                    page,
+                    data: internships,
+                    length: internships.length,
+                    max,
+                });
+            }
+        })
+        .catch((e) => UNPROCESSABLE_ENTITY(next, e));
+};
+
+/**
+ * POST /internship
+ * Used to create a new internship entry
+ */
+export const postInternship = (req: Request, res: Response, next: NextFunction): void => {
+    // @see validator + router
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        return BAD_REQUEST_VALIDATOR(next, errors);
+    }
+
+    const internship: IInternshipEntity = {
+        subject: req.body.subject,
+        description: req.body.description,
+        country: req.body.country,
+        city: req.body.city,
+        address: req.body.address,
+        additional: req.body.additional,
+        isLanguageCourse: req.body.isLanguageCourse,
+        isValidated: req.body.isValidated,
+    };
+
+    Internships.create(internship)
+        .then((created) => res.send(created))
+        .catch((e) => UNPROCESSABLE_ENTITY(next, e));
+};
+
+/**
+ * GET /internship/:id
+ * Used to select a internship by ID
+ */
+export const getInternship = (req: Request, res: Response, next: NextFunction): void => {
+    // @see validator + router
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        return BAD_REQUEST_VALIDATOR(next, errors);
+    }
+
+    Internships.findByPk(req.params.id)
+        .then((val) => {
+            if (checkContent(val, next)) {
+                return res.send(val);
+            }
+        })
+        .catch((e) => UNPROCESSABLE_ENTITY(next, e));
+};
+
+/**
+ * PUT /internship/:id
+ * Used to update internship values
+ */
+export const putInternship = (req: Request, res: Response, next: NextFunction): void => {
+    // @see validator + router
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        return BAD_REQUEST_VALIDATOR(next, errors);
+    }
+
+    Internships.findByPk(req.params.id)
+        .then((internships) => {
+            if (!checkContent(internships, next)) {
+                return undefined;
+            }
+
+            if (req.body.subject) {
+                internships.set('subject', req.body.subject);
+            }
+            if (req.body.description) {
+                internships.set('description', req.body.description);
+            }
+            if (req.body.country) {
+                internships.set('country', req.body.country);
+            }
+            if (req.body.city) {
+                internships.set('city', req.body.city);
+            }
+            if (req.body.address) {
+                internships.set('address', req.body.address);
+            }
+            if (req.body.additional) {
+                internships.set('additional', req.body.additional);
+            }
+            if (req.body.isLanguageCourse) {
```

Comment:
This is a boolean, you will probably have a problem in case it was set to false in the body because the test will tell you that there is no change to apply.

To fix this problem, use the following code:

``` javascript
if (req.body.isLanguageCourse !== undefined) {
    internships.set('isLanguageCourse', req.body.isLanguageCourse ? true : false);
}
```